Tests/src/PublicAPITests/CancellationTokenTests.cpp

```diff
@@ -27,8 +27,7 @@ namespace
 #if RUN_ALL_UNIT_TESTS || RUN_CANCELLATION_TOKEN_TESTS || RUN_CANCELLATIONTOKEN_CONSTRUCTION_TEST
 CSP_PUBLIC_TEST(CSPEngine, CancellationTokenTests, ConstructionAndDestructionTest)
 {
-	csp::common::CancellationToken CancellationToken;
-	// No need to assert anything, just ensuring that construction and destruction don't crash
+	EXPECT_NO_THROW(csp::common::CancellationToken());
```

Comment:
This will only catch exceptions in the constructor and not the destructor. You should use a try-catch with an explicit scope here to guarantee the destructor is called within the try block.

Reply:
It's also worth noting that destructors by default are marked as `nothrow` in c++11 and up, so we should avoid throwing at all costs.

Reply:
Looking at the macro underneath `EXCEPT_NO_THROW` I think we are good as-is, since the try block it expands to is where the statement gets invoked, so both the ctor and dtor will be in there.
![image](https://github.com/magnopus-opensource/connected-spaces-platform/assets/99482500/ab7f2927-ef96-4b8f-826f-2802d3311124)